src/indieweb_utils/posts/discovery.py

```diff
@@ -30,84 +65,59 @@ def discover_original_post(posse_permalink: str) -> str:
+        original_post_url = canonical_links[0].get("href")
+        return original_post_url
 
-        if last_text:
-            last_text = last_text[0].select("p")[-1]
+    # look for text with see original anchor text
 
-            # if permashortlink citation
-            # format = (url.com id)
+    for link in parsed_post.select("a"):
+        if link.text.lower() == "see original".lower() and link.get("href"):
+            original_post_url = link.get("href")
 
-            if re.search(r"\((.*?)\)", last_text.text):
-                permashortlink = re.search(r"\((.*?)\)", last_text.text)
-
-                if permashortlink is not None:
-                    permashortlink_value = "http://" + permashortlink.group(0) + "/" + permashortlink.group(1)
-
-                    candidate_url = permashortlink_value
-                else:
-                    # check for url at end
-                    split_text = last_text.text.split(" ")
-
-                    if split_text[-1].startswith("http://") or split_text[-1].startswith("https://"):
-                        candidate_url = split_text[-1]
-
-        if candidate_url:
-            try:
-                request = requests.get(candidate_url, timeout=5)
-            except requests.exceptions.RequestException:
-                raise PostDiscoveryError("Could not get candidate url")
+            return original_post_url
 
-            parsed_candidate_url = BeautifulSoup(request.text, "lxml")
+    candidate_url = None
 
-            all_hyperlinks = parsed_candidate_url.select("a")
+    last_text = post_h_entry.select(".e-content")
 
-            posse_domain = url_parse.urlsplit(posse_permalink).netloc
+    if last_text:
+        last_text = last_text[0].select("p")[-1]
 
-            for link in all_hyperlinks:
-                if "u-syndication" in link.get("class"):
-                    url_to_check = link.get("href")
+        # if permashortlink citation
+        # format = (url.com id)
 
-                    original_post_url = _syndication_check(url_to_check, posse_permalink, candidate_url, posse_domain)
+        if re.search(r"\((.*?)\)", last_text.text):
+            permashortlink = re.search(r"\((.*?)\)", last_text.text)
 
-                    if original_post_url:
-                        return original_post_url
+        if permashortlink is not None:
+            permashortlink_value = "http://" + permashortlink.group(0) + "/" + permashortlink.group(1)
 
-            all_syndication_link_headers = parsed_post.select("link[rel='syndication']")
+            candidate_url = permashortlink_value
+        else:
+            # check for url at end
+            split_text = last_text.text.split(" ")
 
-            for header in all_syndication_link_headers:
-                if header.get("href") == posse_permalink:
-                    url_to_check = header.get("href")
+            if split_text[-1].startswith("http://") or split_text[-1].startswith("https://"):
```

Comment:
We do this check if a url is http / https in various places. I think we should move the private function `webmentions/discovery.py::_is_http_url` to `utils.py::is_http_url` and reuse it thoughout the project. What do you think?